datarepo-clienttests/src/main/java/common/utils/KubernetesClientUtils.java

```diff
@@ -370,6 +371,7 @@ private static long getApiReadyPods(V1Deployment deployment) throws ApiException
                 pod ->
```

Comment:
There's code duplication here. It looks like these three methods (`getApiPodCount`, `getApiReadyPods`, `printApiPods`) can be refactored as a private method that returns a stream of pods and three methods that operate on that stream.

e.g.,
```
    private static Stream<V1Pod> getPodStream(V1Deployment deployment) throws ApiException {
        String deploymentComponentLabel = deployment.getMetadata().getLabels().get(componentLabel);
        return listPods().stream()
            .filter(
                pod ->
                    deploymentComponentLabel.equals(
                        pod.getMetadata().getLabels().get(componentLabel)));
    }

  private static long getApiPodCount(V1Deployment deployment) throws ApiException {
      return getPodStream(deployment).count();
  }

    private static long getApiReadyPods(V1Deployment deployment) throws ApiException {
        return getPodStream(deployment)
            .filter(pod -> pod.getStatus().getContainerStatuses().get(0).getReady())
            .count();
  }

  public static void printApiPods(V1Deployment deployment) throws ApiException {
    getPodStream(deployment)
        .forEach(p -> logger.debug("Pod: {}", p.getMetadata().getName()));
  }

...
```

Reply:
Yes, great point. Done.